clio-server/src/main/scala/org/broadinstitute/clio/server/service/ReadGroupService.scala

```diff
@@ -26,6 +29,19 @@ class ReadGroupService private (
     )
   }
 
+  def upsertMetadataV2(
+    transferKey: TransferReadGroupV2Key,
+    transferMetadata: TransferReadGroupV1Metadata
+  ): Future[Unit] = {
+    SearchService.upsertMetadata(
+      transferKey,
+      transferMetadata,
+      ReadGroupService.ConverterV2Key,
+      ReadGroupService.ConverterV1Metadata,
```

Comment:
That's the metadataConverter argument.
It's spozed to convert from the incoming metadata upsert to the internal model class.
So I believe what's there is correct.
(-And yeah, this code is insanely finicky, but the types check!-)